### PR TITLE
Added more information about spike generation and precision

### PR DESCRIPTION
**Pull request by jougs**
This fixes trac.783

I propose @steffengraber and @heplesser as reviewers.

### CODE REVIEW

File: extras/userdoc/md/documentation/scheduling-and-simulation-flow.md [hunk 1]
```diff
@@ -77,9 +80,17 @@ wide without need leads to decreased performance due to more update calls and
 communication cycles (small *d<sub>min</sub>*), or increased memory consumption
 of NEST (large *d<sub>max</sub>*).
 
-Related topics
---------------
+## Spike generation and precision
+
+A neuron fires a spike when the membrane potential at the end of an update interval of
+the simulation resolution is above threshold. For most models, the membrane potential
```
Comment (heplesser) [inline extras/userdoc/md/documentation/scheduling-and-simulation-flow.md]: I'd reorder the sentence a little: "A neuron fires a spike when the membrane potential is above threshold at the end of an update interval (i.e., a multiple of the simulation resolution)."

File: extras/userdoc/md/documentation/scheduling-and-simulation-flow.md [hunk 2]
```diff
@@ -77,9 +80,17 @@ wide without need leads to decreased performance due to more update calls and
...
+is then reset to some fixed value and clamped to that value during the refractory time.
+This means that the last membrane potential value at the last time step before the spike
+can vary, while the potential right after the step will usually be the reset potential
+(some models may deviate from this). The time of the spike is always the time at the end
+of the interval during which the threshold was crossed.
```
Comment (heplesser) [inline extras/userdoc/md/documentation/scheduling-and-simulation-flow.md]: I would highlight *the end of the interval*.

File: extras/userdoc/md/documentation/scheduling-and-simulation-flow.md [hunk 3]
```diff
@@ -77,9 +80,17 @@ wide without need leads to decreased performance due to more update calls and
...
-Related topics
---------------
+## Spike generation and precision
+
+A neuron fires a spike when the membrane potential at the end of an update interval of
+the simulation resolution is above threshold. For most models, the membrane potential
+is then reset to some fixed value and clamped to that value during the refractory time.
+This means that the last membrane potential value at the last time step before the spike
+can vary, while the potential right after the step will usually be the reset potential
+(some models may deviate from this). The time of the spike is always the time at the end
```
Comment (heplesser) [inline extras/userdoc/md/documentation/scheduling-and-simulation-flow.md]: Add: "This also means that the membrane potential recording will never show values above the threshold."

File: extras/userdoc/md/documentation/scheduling-and-simulation-flow.md [hunk 4]
```diff
@@ -77,9 +80,17 @@ wide without need leads to decreased performance due to more update calls and
...
+of the interval during which the threshold was crossed.
 
-Please see the [FAQ on precise spike time neurons](../qa-precise-spike-times/index.html)
-for details about neuron update in continuous time and the [documentation on connection management](../connection_management/index.html)
+NEST also has a small number of models that determine the precise time of the threshold
```
Comment (heplesser) [inline extras/userdoc/md/documentation/scheduling-and-simulation-flow.md]: Replace "a small number of" with "some".

**Comment by heplesser** [comment]
@jougs Thanks! I suggested some small changes, with those 👍.

File: extras/userdoc/md/documentation/scheduling-and-simulation-flow.md (hunk 3 above)
Comment (steffengraber) [inline extras/userdoc/md/documentation/scheduling-and-simulation-flow.md]: Standard markdown definition says:

```
A First Level Header
====================

A Second Level Header
---------------------

### Header 3
```

You did it the github way.
So, what is our standard?

**Comment by steffengraber** [comment]
@jougs @heplesser 
We have to decide what we use, standard markdown or the github flavored. 
I used standard markdown.

**Comment by steffengraber** [comment]
@jougs @heplesser 
For me standard means the setext-style.
Github used the atx-style.

Kramdown understand Github-flavored markdown. See  #365.
So, you are right to change the headlines. :+1:

**Comment by jougs** [comment]
@heplesser, @steffengraber: thanks for the review. I hope to have addressed your comments.

**Comment by heplesser** [comment]
ATX-style also seems common in iPython notebooks, so I think it is smart to move all to atx. Thus merging.

**Pull request merged**